src/Query/Provider/DefaultOdm.php

```diff
@@ -54,22 +57,30 @@ public function createQuery(ResourceEvent $event, $entityClass, $parameters)
         $queryBuilder = $this->getObjectManager()->createQueryBuilder();
         $queryBuilder->find($entityClass);
 
-        if (isset($request[$this->getFilterKey()])) {
-            $metadata = $this->getObjectManager()->getMetadataFactory()->getMetadataFor($entityClass);
-            $this->getFilterManager()->filter(
-                $queryBuilder,
-                $metadata,
-                $request[$this->getFilterKey()]
-            );
+        try {
+            if (isset($request[$this->getFilterKey()])) {
+                $metadata = $this->getObjectManager()->getMetadataFactory()->getMetadataFor($entityClass);
+                $this->getFilterManager()->filter(
+                    $queryBuilder,
+                    $metadata,
+                    $request[$this->getFilterKey()]
+                );
+            }
+        } catch (InvalidFieldException $e) {
+            return new ApiProblem(422, $e->getMessage());
```

Comment:
First, we should have `try...catch` inside the `if`. Not sure if we should return `ApiProblem` here. Do we have something like that in other places? Maybe we just should allow the exception to be thrown? It will be catch then here:
https://github.com/zfcampus/zf-rest/blob/master/src/RestController.php#L532-L538

I'm not sure if 422 is also valid response code, we return it always on validation failed and I'd rather don't mix them up - here is different format, without "validation_messages" etc. Please note also that method `createApiProblemFromException` try to get exception code and can use it as response code. Maybe we can use it?